lib/http1.c

```diff
@@ -56,6 +56,15 @@ struct st_h2o_http1_conn_t {
         void *data;
         h2o_http1_upgrade_cb cb;
     } upgrade;
+    struct {
+        unsigned sending : 1;
+        unsigned has_pending_hints : 1;
+        struct {
+            h2o_iovec_t *inbufs;
+            size_t inbufcnt;
+            h2o_send_state_t send_state;
```

Comment:
I can see that we need to buffer the content passed from the generator, now that informational responses could be in flight.

Then, can't we simplify the logic by using the same buffer for storing _any_ pending data, including when we receive an additional 103 while a previous 103 is in flight?

Reply:
@kazuho Thank you for the comment. By this comment I found that `req->res.headers` can be modified while sending 1xx responses, and it'll cause some problems. I fixed that issue at https://github.com/h2o/h2o/pull/1727/commits/bd7e9a13f24b4423a3ab51d0ef448d550803be6a.

OTOH,

> Then, can't we simplify the logic by using the same buffer for storing any pending data, including when we receive an additional 103 while a previous 103 is in flight?

I tried to simplify by buffering final responses too, but it may cause some performance disadvantages, so I left `pending_final` as it is.